### ISSUE TITLE
False positive for unnecessary_lambdas

### ISSUE THREAD

**Issue by kentcb** [dart-lang/linter#1709]
**Describe the issue**
Possibly related to #1561, the following triggers the linter:

```dart
expect(() => builder.build(), ...);
```

However, `expect` takes a `dynamic`, not necessarily a `Function`, so changing to a tear-off changes the behavior to evaluate _before_ invoking `expect`.

**Expected behavior**
No lint.

**Comment by a14n** [comment]
Assuming `expect` takes `dynamic` as first parameter and `builder.build` is a method (not a getter) there are no differences between `expect(() => builder.build(), ...);` and `expect(builder.build, ...);`

**Comment by kentcb** [comment]
You're totally right - I forgot to remove the parentheses when converting to tear-off. Classic. That'll teach me for not just using the IDE. Thanks.
